modules/fundamental/src/advisory/service/mod.rs

```diff
@@ -24,19 +30,79 @@ impl AdvisoryService {
         paginated: Paginated,
         tx: TX,
     ) -> Result<PaginatedResults<AdvisorySummary>, Error> {
+        #[derive(FromQueryResult, Debug)]
+        pub(crate) struct AdvisoryCatcher {
+            pub id: Uuid,
+            pub identifier: String,
+            pub issuer_id: Option<i32>,
+            pub location: String,
+            pub sha256: String,
+            pub published: Option<OffsetDateTime>,
+            pub modified: Option<OffsetDateTime>,
+            pub withdrawn: Option<OffsetDateTime>,
+            pub title: Option<String>,
+            // all of advisory, plus some.
+            pub average_score_cvss3: Option<f64>,
+        }
+
         let connection = self.db.connection(&tx);
 
-        let limiter = advisory::Entity::find().filtering(search)?.limiting(
-            &connection,
-            paginated.offset,
-            paginated.limit,
-        );
+        let cvss3_score: FunctionCall = Func::cust("cvss3_score".into_identity());
+        let cvss3_score = cvss3_score.args([SimpleExpr::Custom("cvss3".to_string())]);
+
+        // To be able to ORDER or WHERE using a synthetic column, we must first
+        // SELECT col, extra_col FROM (SELECT col, random as extra_col FROM...)
+        // which involves mucking about inside the Select<E> to re-target from
+        // the original underlying table it expects the entity to live in.
```

Comment:
This is a little terrifying.

Reply:
Apparently it's normal! We so similar for `SELECT COUNT(*) FROM ( <original select> )` on the limiter stuff.